gcloud/rest/taskqueue/queue.py

```diff
@@ -41,12 +41,128 @@ def headers(self):
+    def ack(self, task):
+        url = '{}/{}:acknowledge'.format(API_ROOT, task['name'])
+
+        body = {
+            'scheduleTime': task['scheduleTime'],
+        }
+
+        with self.google_api_lock:
+            resp = requests.post(url, headers=self.headers(), json=body)
+
+        resp.raise_for_status()
+        return resp.json()
+
+    # https://cloud.google.com/cloud-tasks/docs/reference/rest/v2beta2/projects.locations.queues.tasks/cancelLease
+    def cancel(self, task):
+        url = '{}/{}:cancelLease'.format(API_ROOT, task['name'])
+
+        body = {
+            'scheduleTime': task['scheduleTime'],
+            'responseView': 'BASIC',
+        }
+
+        with self.google_api_lock:
+            resp = requests.post(url, headers=self.headers(), json=body)
+
+        resp.raise_for_status()
+        return resp.json()
+
+    # https://cloud.google.com/cloud-tasks/docs/reference/rest/v2beta2/projects.locations.queues.tasks/delete
+    def delete(self, tname):
+        url = '{}/{}'.format(API_ROOT, tname)
 
-        log.debug('deleting task %s', tid)
         with self.google_api_lock:
             resp = requests.delete(url, headers=self.headers())
 
         resp.raise_for_status()
+        return resp.json()
+
+    # https://cloud.google.com/cloud-tasks/docs/reference/rest/v2beta2/projects.locations.queues.tasks/get
+    def get(self, tname, full=False):
+        url = '{}/{}'.format(API_ROOT, tname)
+        params = {
+            'responseView': 'FULL' if full else 'BASIC',
+        }
+
+        with self.google_api_lock:
+            resp = requests.get(url, headers=self.headers(), params=params)
+
+        resp.raise_for_status()
+        return resp.json()
+
+    # https://cloud.google.com/cloud-tasks/docs/reference/rest/v2beta2/projects.locations.queues.tasks/create
+    def insert(self, payload, tag=None):
+        url = '{}/tasks'.format(self.api_root)
+
+        try:
+            payload = payload.decode()
+        except AttributeError:
+            pass
+        body = {
+            'task': {
+                'pullMessage': {
+                    'payload': payload,
+                    'tag': tag,
+                },
+            },
+            'responseView': 'FULL',
+        }
+
+        with self.google_api_lock:
+            resp = requests.post(url, headers=self.headers(), json=body)
+
+        resp.raise_for_status()
+        return resp.json()
+
+    # https://cloud.google.com/cloud-tasks/docs/reference/rest/v2beta2/projects.locations.queues.tasks/lease
+    def lease(self, num_tasks=1, lease_duration=10, task_filter=None):
+        url = '{}/tasks:lease'.format(self.api_root)
+
+        body = {
+            'maxTasks': min(num_tasks, 1000),
+            'leaseDuration': '{}s'.format(lease_duration),
+            'responseView': 'FULL',
+        }
+        if task_filter:
+            # TODO: this doesn't seem to work?
+            body['filter'] = task_filter
```

Comment:
The task filter doesn't seem to be appended to the body, or the filter isn't applied properly downstream?

Reply:
The filter is not applied. We don't need that capability for v1, so I'll likely leave this as-is.